add support for python f-strings (#4965)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/python-models.md

Referencing [ephemeral](/docs/build/materializations#ephemeral) models is currently not supported (see [feature request](https://github.com/dbt-labs/dbt-core/issues/7288)) 
:::

> Python models also support dynamic referencing of models and configurations within Python f-strings. This allows for more nuanced and dynamic model configurations directly within your Python code. For example:

Author: ```suggestion
<VersionBlock firstVersion="1.8">

From dbt version 1.8, Python models also support dynamic referencing of models and configurations within Python f-strings. This allows for more nuanced and dynamic model configurations directly within your Python code. For example:
```

<File name='models/my_python_model.py'>

```python
# Previously, attempting to access a configuration value like this would result in None
print(f"{dbt.config.get('my_var')}")  # Output before change: None

# Now you can access the actual configuration value
# Assuming 'my_var' is configured to 5 for the current model
print(f"{dbt.config.get('my_var')}")  # Output after change: 5
```

This also means you can use `dbt.config.get()` within Python models, ensuring that configuration values are now effectively retrievable and usable within Python f-strings.

</File>

> 

Author: ```suggestion
</VersionBlock>
```

Referencing [ephemeral](/docs/build/materializations#ephemeral) models is currently not supported (see [feature request](https://github.com/dbt-labs/dbt-core/issues/7288)) 
:::

<VersionBlock firstVersion="1.8">

> From dbt version 1.8, Python models also support dynamic referencing of models and configurations within Python f-strings. This allows for more nuanced and dynamic model configurations directly within your Python code. For example:

Reviewer: I think just configurations, we do not want people to actually reference models in f-strings

<File name='models/my_python_model.py'>

```python
# Previously, attempting to access a configuration value like this would result in None
print(f"{dbt.config.get('my_var')}")  # Output before change: None

# Now you can access the actual configuration value
# Assuming 'my_var' is configured to 5 for the current model
print(f"{dbt.config.get('my_var')}")  # Output after change: 5
```

> This also means you can use `dbt.config.get()` within Python models, ensuring that configuration values are now effectively retrievable and usable within Python f-strings.

Reviewer 2: sugg: so it's more evergreen 

```suggestion
This also means you can use `dbt.config.get()` within Python models to ensure that configuration values are effectively retrievable and usable within Python f-strings.
```

- Accessing the database location of the current model: `dbt.this()` (also: `dbt.this.database`, `.schema`, `.identifier`)
- Determining if the current model's run is incremental: `dbt.is_incremental`

> It is possible to extend this context by "getting" them with `dbt.config.get()` after they are configured in the [model's config](/reference/model-configs). The `dbt.config.get()` method now supports [dynamic access to configurations](#dynamic-configurations) within Python models, enhancing flexibility in model logic. This includes inputs such as `var`, `env_var`, and `target`. If you want to use those values to power conditional logic in your model, we require setting them through a dedicated `.yml` file config:

Author: ```suggestion
It is possible to extend this context by "getting" them with `dbt.config.get()` after they are configured in the [model's config](/reference/model-configs). Starting from dbt v1.8, the `dbt.config.get()` method supports [dynamic access to configurations](#dynamic-configurations) within Python models, enhancing flexibility in model logic. This includes inputs such as `var`, `env_var`, and `target`. If you want to use those values to power conditional logic in your model, we require setting them through a dedicated `.yml` file config:
```

- Accessing the database location of the current model: `dbt.this()` (also: `dbt.this.database`, `.schema`, `.identifier`)
- Determining if the current model's run is incremental: `dbt.is_incremental`

> It is possible to extend this context by "getting" them with `dbt.config.get()` after they are configured in the [model's config](/reference/model-configs). Starting from dbt v1.8, the `dbt.config.get()` method supports dynamic access to configurations within Python models, enhancing flexibility in model logic. This includes inputs such as `var`, `env_var`, and `target`. If you want to use those values to power conditional logic in your model, we require setting them through a dedicated `.yml` file config:

Reviewer 2: sugg: 
- make it more translation friendly
- a lot of code font in the sentence so now there's one less hehe

```suggestion
It is possible to extend this context by "getting" them with `dbt.config.get()` after they are configured in the [model's config](/reference/model-configs). Starting from dbt v1.8, the `dbt.config.get()` method supports dynamic access to configurations within Python models, enhancing flexibility in model logic. This includes inputs such as `var`, `env_var`, and `target`. If you want to use those values for the conditional logic in your model, we require setting them through a dedicated YAML file config:
```

</File>

> #### Dynamic configurations

Author: ```suggestion
<VersionBlock firstVersion="1.8">
#### Dynamic configurations
```

In addition to the existing methods of configuring Python models, you also have dynamic access to configuration values set with `dbt.config()` within Python models using f-strings. This expands the possibilities for custom logic and configuration management.

<File name='models/my_python_model.py'>

```python
def model(dbt, session):
    dbt.config(materialized="table")
    
    # Dynamic configuration access within Python f-strings, which allows for real-time retrieval and use of configuration values.
    # Example: Dynamically accessing a configuration value
    print(f"Dynamic config value: {dbt.config.get('my_var')}")
    # Assuming 'my_var' is set to 5, this will print: Dynamic config value: 5

```

</File>

> 

Author: ```suggestion
</VersionBlock>
```

</File>

<VersionBlock firstVersion="1.8">
    

> #### Dynamic configurations

Reviewer 2: hmm, this H4 isn't being rendered properly in preview: 

![Screenshot 2024-02-22 at 8 42 09 AM](https://github.com/dbt-labs/docs.getdbt.com/assets/107218380/58d85975-0fbb-4c98-807d-ab8361b289ce)

> In addition to the existing methods of configuring Python models, you also have dynamic access to configuration values set with `dbt.config()` within Python models using f-strings. This expands the possibilities for custom logic and configuration management.

Reviewer 2: sugg: i think when using "possibility" , it's generally collocated with "increase"

```suggestion
In addition to the existing methods of configuring Python models, you also have dynamic access to configuration values set with `dbt.config()` within Python models using f-strings. This increases the possibilities for custom logic and configuration management.
```

<File name='models/my_python_model.py'>

```python
def model(dbt, session):
    dbt.config(materialized="table")
    

>     # Dynamic configuration access within Python f-strings, which allows for real-time retrieval and use of configuration values.

Reviewer 2: sugg: making this sentence span 2 lines. that way the horizontal scrollbar doesn't show up. 

imho, h-scrollbars are totally fine if it's code and we can't do anything about it, however, for comments i'd lean towards fewer mouse clicks

>     # Example: Dynamically accessing a configuration value

Reviewer 2: sugg: conciseness, it's repeating the comment before it (ie no real info that's new and of high value) 

```suggestion
```

    print(f"Dynamic config value: {dbt.config.get('my_var')}")

>     # Assuming 'my_var' is set to 5, this will print: Dynamic config value: 5

Reviewer 2: sugg moving this comment to line 258 above
